### ISSUE TITLE
Question: Energy level of the multirotor

### ISSUE THREAD

**Issue by charbelmansour** [microsoft/AirSim#2419]
Hello,
I find that there exist these sensors in Airsim:
-Camera
-IMU
-Magnetometer
-GPS
-Barometer
-Distance sensor
-Lidar
Is there a sensor related to the energy of a multirotor? I mean if there exists a sensor indicating how much energy remains in the multirotor.
If yes, can you tell me please how it works, I mean by what factor does the energy level of the multirotor decrease based on the traveled distance.
Best Regards,
Thank you for your help.

**Comment by rajat2004** [comment]
Currently, there's no battery on energy consumption calculation in AirSim. https://github.com/microsoft/AirSim/issues/672 could be helpful for implementing such a feature

**Comment by aiaarif** [comment]
You can take a look at this repo. They claim to have implemented the energy model. https://github.com/srivatsankrishnan/AirSim-1

**Comment by charbelmansour** [comment]
okay thank you I will check the site.

**Comment by charbelmansour** [comment]
Can you tell me please where I can find rpc-base-reset-plus-energy branch in this repository
https://github.com/srivatsankrishnan/AirSim-1 @aiaarif

**Comment by aiaarif** [comment]
You can switch to different branch by clicking branch icon and selecting the appropriate one. Github should have a nice tutorial on that.

**Comment by charbelmansour** [comment]
Hello @aiaarif 

I have cloned the project and run it on unreal engine. The good news that it works fine.

However, we are running this simulation on Window and the drone is not flying.

Can you please advise us how we can make the drone fly. We have imported the Hello Drone folder but still the drone is not flying.

Thank you for your help,

Charbel

**Comment by aiaarif** [comment]
Please refer to the documentation of Airsim. They have a step by step by guide for that.

**Comment by charbelmansour** [comment]
Hello, I cloned this project: https://github.com/srivatsankrishnan/AirSim-1
I want to get information about the energy of the drone. In this repository, they have implemented a method getTripStats that returns information about the trip including the energy consumed.
I am able to control the drone using python commands.
I am trying to run this code:

import airsim

import numpy as np
import os
import tempfile
import pprint

client = airsim.MultirotorClient(ip="127.0.0.1")
client.confirmConnection()
client.enableApiControl(True)
client.armDisarm(True)

res = client.getTripStats()
print (res.flight_time)


I'm getting the following error:
![ErrorEnergy1](https://user-images.githubusercontent.com/60031311/75200331-18a97800-576e-11ea-9698-d30c5831c2b5.png)



I also tried this code to get the state of the multirotor:
import airsim

import numpy as np
import os
import tempfile
import pprint

client = airsim.MultirotorClient(ip="127.0.0.1")
client.confirmConnection()
client.enableApiControl(True)
client.armDisarm(True)

state = client.getMultirotorState()
s = pprint.pformat(state)
print("state: %s" % s)

I get the following error:
![Error2](https://user-images.githubusercontent.com/60031311/75200584-b56c1580-576e-11ea-94f7-51ad354fc0e3.png)


Any suggestions about the python command that allows to retrieve the trip stats including the energy?

Thank you for your help, 

Best Regards,
Charbel
